src/storage/events/pgboss/move-jobs.ts

```diff
@@ -0,0 +1,126 @@
+  static queueName = 'move-jobs'
+
+  static getQueueOptions(): PgBossQueue {
+    return {
+      name: this.queueName,
+      policy: 'exactly_once',
+    } as const
+  }
+
+  static getWorkerOptions(): WorkOptions {
+    return {
+      includeMetadata: true,
+    }
+  }
+
+  static getSendOptions(payload: MoveJobsPayload): SendOptions {
+    return {
+      expireInHours: 2,
+      singletonKey: `move_${payload.fromQueue}_to_${payload.toQueue}`,
+      singletonHours: 12,
+      retryLimit: 3,
+      retryDelay: 5,
+      priority: 10,
+    }
+  }
+
+  static async handle(job: Job<MoveJobsPayload>) {
+    await multitenantKnex.transaction(async (tnx) => {
+      const resultLock = await tnx.raw('SELECT pg_try_advisory_xact_lock(-5525285245963000611)')
+      const lockAcquired = resultLock.rows.shift()?.pg_try_advisory_xact_lock || false
+
+      if (!lockAcquired) {
+        return
+      }
+
+      const schema = 'pgboss_v10'
+      const fromQueueName = job.data.fromQueue
+      const toQueue = await Queue.getInstance().getQueue(job.data.toQueue)
+
+      if (!toQueue) {
+        logSchema.error(logger, `[PgBoss] Target queue ${job.data.toQueue} does not exist`, {
+          type: 'pgboss',
+        })
+        return
+      }
+
+      try {
+        const sql = `
+            INSERT INTO ${schema}.job (
+                id,
+                name,
+                priority,
+                data,
+                retry_limit,
+                retry_count,
+                retry_delay,
+                retry_backoff,
+                start_after,
+                singleton_key,
+                singleton_on,
+                expire_in,
+                created_on,
+                keep_until,
+                output,
+                policy,
+                state
+            )
+            SELECT 
+                id,
+                '${toQueue.name}' as name,
+                priority,
+                data,
+                retry_limit,
+                retry_count,
+                retry_delay,
+                retry_backoff,
+                start_after,
+                singleton_key,
+                singleton_on,
+                expire_in,
+                created_on,
+                keep_until,
+                output,
+                '${toQueue.policy}' as policy,
+                'created' as state
+            FROM ${schema}.job
+            WHERE name = '${fromQueueName}'
+                AND state IN ('created', 'active', 'retry')
+            ON CONFLICT DO NOTHING
+        `
+
+        await multitenantKnex.raw(sql)
+
+        if (job.data.deleteJobsFromOriginalQueue) {
+          const deleteSql = `
+                DELETE FROM ${schema}.job
+                WHERE name = '${fromQueueName}'
+                    AND state IN ('created', 'active', 'retry')
+            `
+          await multitenantKnex.raw(deleteSql)
+        }
+
+        await multitenantKnex.raw(sql)
```

Comment:
Duplicate of line 106.... Is this intentional? If so why are you running it twice?